### PR TITLE
pandoc-plantuml-filter: init at 0.1.2

### PR DESCRIPTION
**Pull request by CMCDragonkai**
###### Motivation for this change

Adds the Pandoc filter which converts PlantUML code blocks to PlantUML images. Pretty useful!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CMCDragonkai** [comment]
I've tested this with:

```
pandoc -V geometry:margin=1in 'test.md' -o 'test.pdf' --filter pandoc-plantuml
```

It creates a directory called `plantuml-images`.

Note that `plantuml` has to be in the PATH to use this.

I'm not sure how to represent "peer" dependencies in Nix.

File: pkgs/development/python-modules/pandoc-plantuml-filter/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "08673mfwxsw6s52mgglbdz7ybb68svqyr3s9w97d7rifbwvvc9ia";
+  };
+
+  propagatedBuildInputs = [
+    pandocfilters
+  ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:19]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Comment (CMCDragonkai) [reply pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:19]: There are no tests.

How do I use pythonImportsCheck?

Comment (jonringer) [reply pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:19]: import <pkg-name> -> `pythonImportsCheck = [ "<pkg-name>" ];

Comment (CMCDragonkai) [reply pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:19]: There are no modules to import. It only exposes a python script that is executed.

Comment (CMCDragonkai) [reply pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:19]: https://github.com/timofurrer/pandoc-plantuml-filter

**Comment by CMCDragonkai** [comment]
I guess that makes this closer to a python application, not module since there's nothing to import. How to switch to that?

File: pkgs/development/python-modules/pandoc-plantuml-filter/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ buildPythonPackage
+, fetchPypi
+, pandocfilters
+, lib
+}:
+
+buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/development/python-modules/pandoc-plantuml-filter/default.nix:7]:
```suggestion
buildPythonApplication rec {
```

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -7240,6 +7240,8 @@ in {
 
   pandoc-attributes = callPackage ../development/python-modules/pandoc-attributes { };
 
+  pandoc-plantuml-filter = callPackage ../development/python-modules/pandoc-plantuml-filter { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:7243]: move this out of `python-modules` and into the related pkgs dir
```suggestion
  pandoc-plantuml-filter = python3Packages.callPackage ../development/python-modules/pandoc-plantuml-filter { };
```

Comment (CMCDragonkai) [reply pkgs/top-level/python-packages.nix:7243]: Do we keep the alias here or do we change the alias to a top level package?

**Comment by CMCDragonkai** [comment]
Put it next to `pandoc-imagine`. This is ready to be merged.

**Comment by CMCDragonkai** [comment]
@jonringer

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/95143
1 package built:
pandoc-plantuml-filter
```

**Pull request merged**